server/docs.js

```diff
@@ -66,12 +65,43 @@ async function fetchHTMLForId(id, resourceType, req, drive) {
     return fetchHTML(drive, id)
   }
 
-  const {data} = await drive.files.export({
-    fileId: id,
-    // text/html exports are not suupported for slideshows
-    mimeType: resourceType === 'presentation' ? 'text/plain' : 'text/html'
+  try {
+    const {data} = await drive.files.export({
+      fileId: id,
+      // text/html exports are not suupported for slideshows
+      mimeType: resourceType === 'presentation' ? 'text/plain' : 'text/html'
+    })
+
+    return data
+  } catch (e) {
+    // If the Google Drive API returns 403, we fall back to using the export link directly
+    if (e.response.status === 403) {
```

Comment:
HTTP status code 403 stands for `Forbidden`. I am no expert in using Google APIs, but usually it is used to indicate missing authorization. If Google is also using it to indicate "File too large", could this also check the error message or other suitable property, to be sure that we are indeed handling only the expected problem here, and nothing else?

Reply:
Good point, updated [here](https://github.com/reaktor/library/pull/7/commits/d732e72bb6e9626b6b88ac49ca262c74cfe9d837)